frontend/src/metabase/dashboard/hooks/use-dashboard-url-params.unit.spec.tsx

```diff
@@ -0,0 +1,211 @@
+import type { Location } from "history";
+import { Route } from "react-router";
+
+import { renderWithProviders, screen } from "__support__/ui";
+
+import { useDashboardUrlParams } from "./use-dashboard-url-params";
+
+describe("useDashboardUrlParams", () => {
+  describe("should handle backward compatibility when `theme=transparent` and `background=false` (metabase#43838)", () => {
+    it('should return theme: "light" by default', async () => {
+      interface ComponentProps {
+        location: Location;
+      }
+      const Component = ({ location }: ComponentProps) => {
+        const { theme } = useDashboardUrlParams({
+          location,
+          onRefresh: jest.fn(),
+        });
+        return <>theme: {theme}</>;
+      };
+
+      renderWithProviders(
+        <>
+          <Route path="*" component={Component} />
+        </>,
+        {
+          withRouter: true,
+        },
+      );
+
+      expect(await screen.findByText("theme: light")).toBeInTheDocument();
+    });
+
+    it('should return theme: "light" when provided', async () => {
+      interface ComponentProps {
+        location: Location;
+      }
+      const Component = ({ location }: ComponentProps) => {
+        const { theme } = useDashboardUrlParams({
+          location,
+          onRefresh: jest.fn(),
+        });
+        return <>theme: {theme}</>;
+      };
+
+      renderWithProviders(
+        <>
+          <Route path="*" component={Component} />
+        </>,
+        {
+          withRouter: true,
+          initialRoute: "#theme=light",
```

Comment:
What if we expose the `initialRoute` field in the `setup(...)` function? I think that would cover all of the use cases here, and reduce the code duplication.

Reply:
Fixed in https://github.com/metabase/metabase/pull/45214/commits/f9de64d76c539b0007561d6c783a5ce9b798635a